### PR TITLE
Fixed Visual Mode + 'S' Not Working (VSCodeVim #4072)

### PR DESCRIPTION
**Pull request by micklett**
Fixed Visual Mode + 'S' Not Working (VSCodeVim #4072)

- Combined C and R commands in Visual Mode since they had the same code
- Added a command for S to delete highlighted lines in visual mode
- Added check to disable S when using sneak or surround
- Pressing C, R, or S now changes to insert mode

**What this PR does / why we need it**:
Fix the issue reported, 

1. Open the file and focus to some text.
2. Click `v` to enter visual mode.
3. Click `jj` to select two lines down.
4. Click `S` to attempt to delete the selected lines.
5. Click e to see that an 'e' has been inserted into the location where you started and ended the selection.

Also changed it so that `C`, `R`, and `S` all put the user in Insert Mode

**Which issue(s) this PR fixes**
#4072 

**Special notes for your reviewer**:
Tested locally and tried to write a test; unfortunately, since vim.surround is on by default, `S` has a different behavior (as it should when vim.surround is on), and so testing will not pass.

Also, VSCode automatically changes the arrays to multiple lines. If this is a problem, I can try to amend my last commit with the arrays on the same line.

### CODE REVIEW

**Comment by J-Fields** [comment]
Thanks for the PR!

> Also, VSCode automatically changes the arrays to multiple lines. If this is a problem, I can try to amend my last commit with the arrays on the same line.

Don't worry about it, we use prettier to keep everything consistent and it likes to do that.

> Tested locally and tried to write a test; unfortunately, since vim.surround is on by default, S has a different behavior (as it should when vim.surround is on), and so testing will not pass.

You can modify the config for a test [like this](https://github.com/VSCodeVim/Vim/blob/master/test/mode/modeNormal.test.ts#L13).

**Comment by micklett** [comment]
OK, I've added a test and updated a few others. Sorry about the strange commits; trying to get my local branch up to date with the recent changes.

**Comment by J-Fields** [comment]
Awesome, thanks! Sorry for the mayhem caused by the refactor hahaha

File: test/mode/modeVisual.test.ts [hunk 1]
```diff
@@ -1222,7 +1237,7 @@ suite('Mode Visual', () => {
         'ggvep'.split('') // replace 'replace'
       );
       await modeHandler.handleMultipleKeyEvents(['g', 'v']);
-
+
```
Comment (J-Fields) [inline test/mode/modeVisual.test.ts]: Run `gulp forceprettier`, please

**Comment by J-Fields** [comment]
Any reason this doesn't also extend support to visual line and visual block modes?

**Comment by J-Fields** [comment]
Hey @micklett, do you plan on coming back to this PR?

**Pull request closed without merging**